UT4MasterServer/Services/StatisticsService.cs

```diff
@@ -490,5 +492,36 @@ private static Statistic MergeStatistics(List<Statistic> statistics)
 		return merged;
 	}
 
+	/// <summary>
+	/// This method will delete statistics that are older than X <paramref name="days"/>
+	/// </summary>
+	/// <param name="days">Number of days that should be kept for statistic records. It shouldn't be less than 30 since they are used for monthly statistics calculation.</param>
+	/// <param name="skipFlagged">When set to <see langword="false" /> it will also delete flagged statistics. Flagged statistics should be deleted only after they are inspected.</param>
+	/// <returns>Number of deleted statistic records</returns>
+	public async Task<long> DeleteOldStatisticsAsync(int days = 30, bool skipFlagged = true)
+	{
+		logger.LogInformation("Deleting statistics that are older than: {Days}, {Flagged}.", days, skipFlagged);
+
+		if (days < MinimumDaysForDeletingOldStatistics)
+		{
+			throw new ArgumentException($"You can delete only statistics that are at least {MinimumDaysForDeletingOldStatistics} days old.", nameof(days));
+		}
+
+		var removeBeforeDate = DateTime.UtcNow.Date.AddDays(-days);
+
+		var filter = Builders<Statistic>.Filter.Lt(f => f.CreatedAt, removeBeforeDate);
+		
+		if (skipFlagged)
+		{
+			filter &= Builders<Statistic>.Filter.Eq(f => f.Flagged, null);
```

Comment:
This works? Either way, I would prefer the use of `Filter.Exists`.

Reply:
It works, but I will replace it with Exists.